src/createServer.js

```diff
@@ -1,3 +1,52 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const { convertToCase } = require('./convertToCase/convertToCase');
+const http = require('http');
+
+const cases = ['SNAKE', 'KEBAB', 'CAMEL', 'PASCAL', 'UPPER'];
+
+function createServer() {
+  return http.createServer((req, res) => {
+    const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+    const originalText = normalizedUrl.pathname.slice(1);
+    const targetCase = normalizedUrl.searchParams.get('toCase');
+    const errors = [];
+
+    if (!originalText) {
+      errors.push({ message: 'Text to convert is required. '
+        + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".' });
+    }
+
+    if (targetCase === null) {
+      errors.push({ message: '"toCase" query param is required. '
+        + 'Correct request is: "/<TEXT_TO_CONVERT>?toCase=<CASE_NAME>".' });
+    } else if (!cases.includes(targetCase)) {
+      errors.push({ message: 'This case is not supported. '
+        + 'Available cases: SNAKE, KEBAB, CAMEL, PASCAL, UPPER.' });
```

Comment:
It's good practice to use template literals like this. It doesn't require to escape special characters (text inside will be as it is including spacing). Also, you can include variables inside. You have the variable cases, and use values from it to have always up-to-date errors

```suggestion
      errors.push({ message: `This case is not supported. Available cases: ${cases.join(', ')}.`});
```